beacon_node/websocket_server/src/lib.rs

```diff
@@ -67,31 +61,26 @@ pub fn start_server<T: EthSpec>(
     let broadcaster = server.broadcaster();
 
     // Produce a signal/channel that can gracefully shutdown the websocket server.
-    let exit_channel = {
-        let (exit_channel, exit) = tokio::sync::oneshot::channel();
-
-        let log_inner = log.clone();
-        let broadcaster_inner = server.broadcaster();
-        let exit_future = async move {
-            let _ = exit.await;
-            if let Err(e) = broadcaster_inner.shutdown() {
-                warn!(
-                    log_inner,
-                    "Websocket server errored on shutdown";
-                    "error" => format!("{:?}", e)
-                );
-            } else {
-                info!(log_inner, "Websocket server shutdown");
-            }
-        };
-
-        // Place a future on the handle that will shutdown the websocket server when the
-        // application exits.
-        tokio::spawn(exit_future);
-
-        exit_channel
+    let exit = executor.exit();
+    let log_inner = log.clone();
+    let broadcaster_inner = server.broadcaster();
+    let exit_future = async move {
+        let _ = exit.await;
```

Comment:
Again here. Is there a race condition between this firing and the global select?

Reply:
There's just one exit future here as we are doing `runtime_handle().spawn()`. Since we are running websockets on a `std::thread`